documents/guides/local_setup_ubunutu_wsl_from_scratch.md

```diff
@@ -0,0 +1,122 @@
+# Local Teiserver setup
+
+Using WSL with Ubuntu 22.0.4 downloaded from the windows store
+
+Installed git, curl, and unzip
+```bash
+sudo apt install git curl unzip
+```
+git and curl were already pre-installed with their latest available versions.
+
+Installed [asdf](https://github.com/asdf-vm/asdf) for managing versinos of elixir and erlang:
+```bash
+git clone https://github.com/asdf-vm/asdf.git ~/.asdf --branch v0.14.0
+```
+
+Set up asdf in bashrc:
+```bash
+echo '
+. "$HOME/.asdf/asdf.sh"
+. "$HOME/.asdf/completions/asdf.bash"' >> ~/.bashrc
+```
+
+Add asdf plugins for elixir and erlang:
+```bash
+asdf plugin-add elixir https://github.com/asdf-vm/asdf-elixir.git
+asdf plugin add erlang https://github.com/asdf-vm/asdf-erlang.git
+```
+
+Install dependencies for erlang install:
+```bash
+sudo apt-get -y install build-essential autoconf m4 libncurses5-dev libwxgtk3.0-gtk3-dev libwxgtk-webview3.0-gtk3-dev libgl1-mesa-dev libglu1-mesa-dev libpng-dev libssh-dev unixodbc-dev xsltproc fop libxml2-utils libncurses-dev openjdk-11-jdk
+```
+
+Install elixir and erlang versions tracked in project (must be done from root of project)
+```bash
+asdf install
+```
+
+Get and compile local elixir dependencies (say yes to prompts for additional installs):
+```bash
+mix deps.get
+mix deps.compile
+```
+
+Install postgres:
+```bash
+sudo apt install postgresql
+```
+
+Set password for postgres user (account should have been created automatically by the apt install)
+```bash
+sudo passwd postgres
```

Comment:
What happens if you skip this step? Is the default user/password just postgres postgres?

Reply:
I'm not sure what the default password is. None of the few that I guessed were correct.

Reply:
Changing postgres password that way is not recommended, see https://serverfault.com/a/325596 for more details.

Reply:
> I'm not sure what the default password is. None of the few that I guessed were correct.

The reason I asked the question originally is because based on my own notes I didn't have to do this step. You can login to the db using `teiserver_dev WITH PASSWORD '123456789'`. So not sure what setting the postgres password was trying to accomplish.

Reply:
> I'm not sure what the default password is. None of the few that I guessed were correct.

> I'm not sure what the default password is. None of the few that I guessed were correct.

The reason I asked the question originally is because based on my own notes I didn't have to do this step. You can login to the db using `teiserver_dev WITH PASSWORD '123456789'`. So not sure what setting the postgres password was trying to accomplish.

Reply:
Thanks for pointing this out. I don't think I ever actually needed to log in to the postgres user account directly, so leaving this off and making it to the user can't be logged into makes sense.

Although, since this is specific to a WSL setup, the risks of having the user account have a password and available for login is much less relevant than it would be for a live server or cloud instance.